pkg/kubectl/resource/builder.go

```diff
@@ -806,12 +806,13 @@ func HasNames(args []string) (bool, error) {
 
 // MultipleTypesRequested returns true if the provided args contain multiple resource kinds
 func MultipleTypesRequested(args []string) bool {
+	if len(args) == 1 && args[0] == "all" {
+		return true
```

Comment:
@janetkuo 
> After this call, args has been changed (aliases, including all, are expanded).

Apologies, I see what you mean now. I had tested the values of `args` in a line located above the call to `ResourceTypeOrNameArgs` in [this PR](https://github.com/kubernetes/kubernetes/pull/39042#discussion_r96523358), and had not realized that its value is mutated by the time it is used in 
```go
if resource.MultipleTypesRequested(args) || cmdutil.MustPrintWithKinds(objs, infos, sorter) {
```

> Would you explain more on when the following situation would happen?
>>arguments from a sub command are passed to it, without first going through other functions in resource/builder.go

I realize now that, at least for the time being, `MultipleTypesRequested` will never receive `all` as one of its arguments, however a potential situation where it will is if a call to this helper is ever made before a call to `ReplaceAliases`.